app/models/hike.rb

```diff
@@ -77,13 +77,8 @@ def average_rating_sql
   end
 
   def self.top_rated
-    review = Review.find_by_sql(
-      'SELECT AVG("reviews"."rating") AS average_rating,
-        hike_id AS hike_id FROM "reviews"
-        GROUP BY "reviews"."hike_id"
-        ORDER BY average_rating DESC
-        LIMIT 1'
-    )
+    top_rated = Review.select("AVG(reviews.rating) as average_rating, reviews.hike_id").group(:hike_id).order("average_rating DESC").limit(1)
+    review = [top_rated[0]]
```

Comment:
i don't think this is necessary, you should already have a collection with only one item because you used a `limit(1)`.

Furthermore if the query has zero results the next line will never execute the first part of the ternary clause.

```
[17] pry(main)> review = [[][0]]
=> [nil]                                      ^
[18] pry(main)> review.empty? ? 'this will never be called' : 'this is a false positive'
=> "this is a false positive"
```